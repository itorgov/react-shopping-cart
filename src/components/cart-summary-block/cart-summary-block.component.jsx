import React from 'react';

import {CartTotal} from '../cart-total/cart-total.component';
import {Checkbox} from '../checkbox/checkbox.component';
import {Button} from '../button/button.component';

import styles from './cart-summary-block.module.scss';
import {ReactComponent as IconLock} from './lock.svg';

export const CartSummaryBlock = () => (
    <div className={styles.summary}>
        <CartTotal/>
        <p>Shipping & taxes will calculated at checkout</p>
        <Checkbox>
            I agree to <a href='/'>Terms & Conditions</a>
        </Checkbox>
        <Button variant='black'>
            Checkout
            <IconLock/>
        </Button>
    </div>
);
