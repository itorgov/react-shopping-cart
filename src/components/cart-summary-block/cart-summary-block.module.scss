.summary {
  flex: 1;

  padding: 6rem 10rem 6rem 3rem;
  background-color: var(--color-primary);

  @media only screen and (max-width: 56.25em) {
    padding: 3rem;
  }

  & p {
    font-size: 1.3rem;
  }
}
