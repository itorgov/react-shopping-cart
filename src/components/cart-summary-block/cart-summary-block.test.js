import React from "react";
import {shallow} from "enzyme";

import {CartSummaryBlock} from './cart-summary-block.component';

describe('<CartSummaryBlock />', () => {
    it('expect to render', () => {
        expect(shallow(<CartSummaryBlock />)).toMatchSnapshot();
    });
});
