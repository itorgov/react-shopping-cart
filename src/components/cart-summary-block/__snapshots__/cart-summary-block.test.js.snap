// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CartSummaryBlock /> expect to render 1`] = `
<div
  className="summary"
>
  <CartTotal />
  <p>
    Shipping & taxes will calculated at checkout
  </p>
  <Checkbox>
    I agree to 
    <a
      href="/"
    >
      Terms & Conditions
    </a>
  </Checkbox>
  <Button
    variant="black"
  >
    Checkout
    <ForwardRef(SvgLock) />
  </Button>
</div>
`;
